packages/admin/components/forms/Fieldset/Fieldset.styles.ts

```diff
@@ -11,10 +11,15 @@ export const Fieldset = styled.fieldset`
   flex-basis: var(--form-grid-item-width-wide);
 `;
 
-export const Legend = styled.legend`
-  padding-block-end: var(--fieldset-field-spacing, ${pxToRem(24)});
+export const Legend = styled.legend<LegendProps>`
+  --fieldset-field-spacing: ${({ $noGap }) => ($noGap ? 0 : pxToRem(24))};
```

Comment:
Okay to set `--fieldset-field-spacing` here?

Reply:
Unfortunately, this would mean that `--fieldset-field-spacing` can't be defined by a parent component, because you're redefining it here in Fieldset. If a parent component needs the gap to be 0, it should set `--fieldset-field-spacing: 0`.

Reply:
Side note: I make this mistake a lot too. 😅 I think we should define our variables in `root.ts`, and keep the naming structure the same (`--[component-name]-[property]`). That way we have a central location for all of the CSS variables that are used in the app. @dananjohnson , what do you think?

Reply:
I'm not sure I understand what you're defining in `root.ts`. You'd have a bunch of `--componentName-property` defined at `:root {}` there? I do like the naming convention. I might even match the casing of the CSS property to the component (i.e. `--Fieldset-field-spacing`) to more clearly indicate the relationship.